### PR TITLE
modem: modem_cmux: Quality improvements

### PR DESCRIPTION
**Pull request by bjarki-andreasen**
This PR provides a handful of improvements.

### CODE REVIEW

File: subsys/modem/modem_cmux.c [hunk 1]
```diff
@@ -452,7 +533,7 @@ static void modem_cmux_on_dlci_frame(struct modem_cmux *cmux)
 		break;
 
 	default:
-		modem_cmux_log_unknown_frame(cmux);
+		LOG_WRN("Unknown DLCI frame type");
```
Comment (ycsin) [inline subsys/modem/modem_cmux.c:536]:
```suggestion
		LOG_WRN("Unknown %s frame type", "DLCI");
```

File: subsys/modem/modem_cmux.c [hunk 2]
```diff
@@ -373,7 +449,7 @@ static void modem_cmux_on_control_frame(struct modem_cmux *cmux)
 		break;
 
 	default:
-		modem_cmux_log_unknown_frame(cmux);
+		LOG_WRN("Unknown control frame type");
```
Comment (ycsin) [inline subsys/modem/modem_cmux.c:452]:
```suggestion
		LOG_WRN("Unknown %s frame type", "control");
```

**Review by ycsin — COMMENTED** [review]
not really a thorough review, just spotted some potential bytes savings

**Review by rerickson1 — DISMISSED** [review]

**Comment by bjarki-trackunit** [comment]
Added optimizations requested by @ycsin, and created a log entry specifically for the `MODEM_CMUX` module since it produces an immense amount of log output

**Review by ycsin — APPROVED** [review]
Ideally the commit to optimize the LOG messages should be squashed with the one that introduced them.

I would probably also omit the _"This commit ..."_ in all the commit messages, they seem somewhat redundant

**Review by rerickson1 — APPROVED** [review]

**Pull request merged**